source/server/admin/stats_handler.cc

```diff
@@ -91,74 +92,41 @@ Http::Code StatsHandler::handlerStats(absl::string_view url,
   absl::Status histogram_buckets_status =
       Utility::histogramBucketsParam(params, histogram_buckets_mode);
   if (!histogram_buckets_status.ok()) {
-    response.add(histogram_buckets_status.message());
-    return Http::Code::BadRequest;
+    return Admin::makeStaticTextRequest(histogram_buckets_status.message(), Http::Code::BadRequest);
   }
 
   const absl::optional<std::string> format_value = Utility::formatParam(params);
   bool json = false;
   if (format_value.has_value()) {
     if (format_value.value() == "prometheus") {
-      return handlerPrometheusStats(url, response_headers, response, admin_stream);
+      Buffer::OwnedImpl response;
+      Http::Code code = prometheusStats(path, response);
+      return Admin::makeStaticTextRequest(response, code);
```

Comment:
No that needs to be a follow-up. It would be awesome if someone who uses Prometheus could take that on :)

The challenge there will be that I think the sort-order in Prometheus will be based on tagExtractedName() rather than name(), but we might be able to factor that out and share most of the code.

My hope is that this will mostly be a new Render implementation, but it might be more than that due to the more complex structure of the output.

Reply:
You can tag me on that TODO. I wrote a lot of the current prometheus code.